CMakeLists.txt

```diff
@@ -347,8 +347,17 @@ target_compile_options(
 
 enable_testing()
 
-# EXCLUDE_FROM_ALL so 'make install' doesn't attempt installation of googletest
-#add_subdirectory(external/googletest EXCLUDE_FROM_ALL)
+# Install libReactiveSocket.a and yarpl headers normally.
+install(TARGETS ReactiveSocket DESTINATION lib)
+install(DIRECTORY yarpl/include/yarpl DESTINATION include FILES_MATCHING PATTERN "*.h")
+
+# RSocket headers get installed in $INSTALL_DIR/src/Foo.h initially, because
+# they use #includes of the form "src/RSocketServer.h".  We also need them to be
+# $INSTALL_DIR/rsocket/Foo.h.
+install(DIRECTORY src DESTINATION include FILES_MATCHING PATTERN "*.h")
+install(CODE "execute_process(COMMAND rm -r ${CMAKE_INSTALL_PREFIX}/include/rsocket)")
+install(CODE "execute_process(COMMAND cp -r ${CMAKE_INSTALL_PREFIX}/include/src ${CMAKE_INSTALL_PREFIX}/include/rsocket)")
```

Comment:
Probably, but now that I've tried this hacky `src/` copy I realize it doesn't quite work... It's far too easy to transitively include both `"src/RSocketServer.h"` and `"rsocket/RSocketServer.h"`, leading to duplicate definition compiler errors.

I think I'm going to try to rename `src/` to `rsocket/` and then fix up all the #includes in one go before I ship this.